Cut 0.19.0 release.

dhermes opened the pull request:
This is the first release after the rename.

Also updated the account that does deployment from Silvio's to one shared by google-cloud. We'll likely need to ditch the `deploy` section and write a more customized script using `twine` once we deploy versions for each sub-package.

Closes #2264.

jonparrott commented:
@dhermes qq: did we resolve the namespace/import issue? Will users be able to do;

```python
from google.cloud import storage

storage_client = storage.Client()
```
?

dhermes commented:
Other than the services in #2258 (`logging` and `pubsub`) everything is a normal package (not a namespace package)

jonparrott commented:
Can I be a jerk and block the release on that issue? I would prefer not to have to update sample code twice.

dhermes commented:
@bjwatson Thoughts on timing?

@jonparrott We are just in a state of flux until a release is cut.

jonparrott commented:
@dhermes how bad is the state of flux? is it better to push this out quickly or wait for #2258 to be resolved? Are they any other issues that would make upgrading more trivial than just updating imports?

(Apologies for the slew of questions, I just want to make sure I understand the full picture)

jonparrott commented:
@dhermes also, do we have a changelog?

dhermes commented:
I have yet to make the changelog. I'm a bit weary of doing it, it'll be big. Also some people may have started doing squash merges so my typical script won't see those PRs?

bjwatson commented:
@dhermes @jonparrott I would like to resolve the following first if possible:

* https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2258
* https://github.com/GoogleCloudPlatform/google-cloud-python/issues/1893#issuecomment-245410626 (issue Jacob raised about gRPC dependency remaining in `google/longrunning/operations_pb2.py`)
* https://github.com/googleapis/gax-python/pull/127 (LGTM'd, but need to be merged and released, and the gax and gcloud dependencies updated)

I can be talked out of any of these if there are urgent overriding needs to get this release out.

dhermes commented:
@bjwatson Do you have a timeframe on this? e.g. why hasn't the LGTM'd PR been merged?

bjwatson commented:
@dhermes 
* https://github.com/GoogleCloudPlatform/google-cloud-python/issues/1893#issuecomment-245410626: this will be done today; it's taken much longer than expected due to the many (albeit valuable) discussions about how best to manage the gRPC dependencies
* https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2258: realisitically, I won't be able to get this done before Tuesday or Wednesday of next week, unless @geigerj can take this sooner
* https://github.com/googleapis/gax-python/pull/127: @geigerj can you merge and publish this today?

geigerj commented:
@bjwatson gax-python 0.13.0 has been published, although until we refresh codegen, the `gax-google-...` packages will still be restricted to 0.12.x.

bjwatson commented:
We can refresh the packages as part of https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2258, unless @dhermes needs this done sooner.

dhermes commented:
It'd be nice to have it in today, but I trust you're going as quick as you can.

bjwatson commented:
@dhermes I am. Regarding https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2258, we can reduce risk and maybe get it done sooner by just focusing on the GAX namespace.

The other part of the problem is fixing the gRPC namespace, mainly to be more consistent, but also to avoid conflicts for the ML libraries (the gRPC packages currently use `google.cloud.{api}` as a namespace for language, vision, and speech). We can hand-fix the ML gRPC packages ASAP, and then fix the other gRPC packages a little later.

dhermes commented:
That SGTM.

overset commented:
This would be great to get a 0.19.0 onto PyPy since the Datastore v1beta2 API is being deprecated on 9/30

https://cloud.google.com/datastore/release-notes
> This is a follow-up to our recent email to remind you that Google Cloud Datastore is turning down the v1beta1 and v1beta2 API on September 30, 2016.

The previous 0.18.1 release had `v1beta2`.

dhermes commented:
@overset Thanks for the heads up. FWIW, we haven't supported `v1beta2` for a very long time.

- We upgraded from `v1beta3` to `v1` on August 16, 2016 (https://github.com/GoogleCloudPlatform/google-cloud-python/commit/c52d3635c16b015a29cb29a39de7a1275270e15a).
- We upgraded from `v1beta2` to `v1beta3` on January 20, 2016 (https://github.com/GoogleCloudPlatform/google-cloud-python/commit/64c28fdee0e053f1fc5443a88b0195cd1f72a0ed)

Check out the release history for relevant releases near those dates

overset commented:
@dhermes My previous comment was flat-out wrong and thanks for confirming 0.18.1 having upgdaded to Datastore `v1beta3` via  https://github.com/GoogleCloudPlatform/google-cloud-python/commit/64c28fdee0e053f1fc5443a88b0195cd1f72a0ed. We haven't seen a turn-down notice for v1beta3 but it is marked as deprecated. (Thanks for the `s/PyPy/PyPI/` on the previous comment as well)

dhermes commented:
Sure thing. Not sure when `v1beta3` will be turned off.

daspecster commented:
@dhermes @tseaver @bjwatson, unless I'm missing something, I think we have all the items for this done?

We might want to have a PR to switch the naming back from https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2264 or add it to this PR.

dhermes commented:
Time to make release notes. Yay for 130+ PRs!

```
$ git log -1 --pretty=%H
6fb2d73ff6350590aa061bcd902422828340f337
$ git log 0.18.1..HEAD | grep 'Merge pull request ' |  awk -F ' ' '{print $4}' | sort
```

yields 

#2035
#2039
#2050
#2051
#2052
#2054
#2056
#2058
#2060
#2062
#2065
#2071
#2078
#2081
#2082
#2085
#2086
#2087
#2088
#2089
#2090
#2094
#2095
#2096
#2097
#2099
#2100
#2103
#2106
#2107
#2109
#2110
#2115
#2122
#2124
#2129
#2135
#2139
#2141
#2147
#2148
#2149
#2150
#2151
#2153
#2155
#2161
#2162
#2164
#2165
#2167
#2171
#2172
#2174
#2175
#2178
#2179
#2181
#2187
#2188
#2189
#2191
#2195
#2197
#2199
#2202
#2203
#2210
#2211
#2212
#2213
#2214
#2215
#2219
#2223
#2224
#2227
#2235
#2236
#2237
#2240
#2241
#2244
#2245
#2247
#2248
#2249
#2250
#2251
#2252
#2254
#2257
#2260
#2261
#2262
#2265
#2267
#2272
#2273
#2276
#2279
#2280
#2283
#2284
#2286
#2288
#2290
#2292
#2293
#2295
#2299
#2302
#2303
#2304
#2306
#2307
#2308
#2310
#2311
#2319
#2320
#2321
#2323
#2331
#2333
#2334
#2335
#2336
#2338
#2339
#2345

dhermes commented:
I also used the [UI][1] to look through merged PRs since August 9. I realized part way through that `0.18.1` wasn't a real release, it was a patch release due to broken PyPI deps, so I wanted to be extra careful.

Here is the release notes with stuff left out after. @daspecster Feel free to edit the `vision` parts inline as much as you like

-----

## Core

- Renaming `gcloud` package as `google.cloud` and moving on
  PyPI from `gcloud` to `google-cloud` (#2223, #2334, #2264, #2284)
- Dropped support for Python 2.6 (#2051)
- Upgrading to GAX 0.13.0 (#2324, #2336)
- Removing gRPC / GAX restriction to Python 2.7 and removing the
  `gcloud[grpc]` extra. (#2308, #2203)
- Add missing exception class for '502 Bad Gateway' response code (#2078)
- Bulk spelling / style / grammar docs fix (#2139)
- Upgrading all usage of gRPC from `beta` to stable (#2149)
- Adding shared `Operation` class for long-running operations (#2165)
- Renaming core docs: `gcloud-api` to `google-cloud-api`, `gcloud-auth`
  to `google-cloud-auth` and `gcloud-config` to `google-cloud-config` (#2280)
- Modify user agent metadata for gRPC (#2286)

## Resource Manager

- Add alpha warning to docs (#2345)

## Datastore

- Upgrade from `v1beta3` to `v1` (#2107)
- Add gRPC support (#2097, #2099)
- **BREAKING CHANGE**: Making batches and transactions more robust to
  failure (#2303, fixes #2297). Now a `Batch` must call `begin()`
  before adding mutations.
- Adding emulator support for gRPC implementation (#2244)

## BigQuery

- Add partition support (#2058)
- Fix docs error about starting a job (#2338)
- Add `QueryJob.results` method (#2321)
- Adding desiganted `schema` module (#2060)
- Fixing docs typo about destination table (#2250)
- Update state values in usage doc (#2251)
- Adding `list_projects` method (#2254)
- Adding configuration properties for `maximumBillingTier` and
  `maximumBytesBilled` for `QueryJobs` (#2276)
- Support parsing `DATE` columns from Standard SQL tables (#2265)

## Vision

- Initial implementation (#2320, #2319, #2299, #2237, #2148, #2167, #2179, #2224, #2235, #2236)

## Pub / Sub

- Re-enable gRPC / GAX by default if it is importable. (#2085)
- Fix bug mapping publish timestamp to result in
  gRPC / GAX implementation (#2262)
- Adding emulator support for gRPC / GAX implementation (#2245)

## Logging

- Re-enable gRPC / GAX by default if it is importable. (#2085)
- Map gRPC exceptions onto exceptions from
  `google.cloud.exceptions` (#2094)
- Product rename (Stackdriver) (#2273)
- Docs bug: `filter` rename to `filter_` (#2279)

## Bigtable

- Moved HappyBase into its own [project][1] (#2109)
- Removing Bigtable undelete (not present in `v2`) (#2210)
- Adding support for Bigtable emulator (#2249)

## Monitoring

- Add `Group` support (#2035)
- Add Factory Methods for Metric/Resource/Timeseries (#2071)
- Fix `ValueType` link in docs (#2150)
- Add support for comparison filters (#2234)
- Add docs in README (#2054)

## Natural Language

- Initial implementation (#2062, #2162, #2164, #2172, #2187, #2189, #2211)

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python-happybase

-----

Not in notes.
Misc: #2339, #2335, #2311, #2065, #2067, #2085, #2088, #2090, #2096, #2097, #2135, #2141, #2151, #2212, #2227, #2241, #2257, #2272, #2288, #2292, #2304, #2056
Docs stuff (some dead code): #2052, #2103, #2110, #2039, #2115, #2129, #2122, #2155, #2147, #2171, #2175, #2178, #2188, #2191, #2174, #2199, #2153, #2219, #2247, #2267, #2269, #2282, #2281, #2302
Refactor private helpers: #2082, #2124, #2240
Misc. System test changes: #2313, #2050, #2087, #2202
Generated code: #2081
System test robustness: #2333, #2293, #2331, #2248, #2323, #2307, #2310, #2306, #2089, #2100, #2086, #2106, #2095, #2161, #2181, #2195, #2197, #2214, #2213, #2215, #2260, #2261, #2252, #2283, #2290, #2295 

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc

tseaver commented:
I've tweaked the entries all into imperative mode.

dhermes commented:
Thanks @tseaver

daspecster commented:
I updated vision to be more verbose to it's easier to reference later if needed.

dhermes commented:
I just used that exact key locally and had no issues:

```
TRAVIS_BUILD_ID=161085138 tox -e system-tests -- logging
```

I restarted https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/161085138, hopefully it will go green

dhermes commented:
@tseaver @daspecster Are we good to merge here? (Don't know why Travis is being fast on "push" builds to master but not "pr" builds.)

daspecster commented:
Green!

tseaver commented:
LGTM

dhermes commented:
Pushed the tag and added the notes: https://github.com/GoogleCloudPlatform/google-cloud-python/releases/tag/0.19.0 (may need #2347 update)

Push to PyPI should come from: https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/161121136

tseaver commented:
I updated the release notes to include the changes from #2347.
